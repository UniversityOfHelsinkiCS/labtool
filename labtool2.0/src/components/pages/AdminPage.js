import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Button, Loader } from 'semantic-ui-react'
import { Redirect } from 'react-router'
import { getAllCI } from '../../services/courseInstance'
import { resetLoading } from '../../reducers/loadingReducer'
import { getAllUsers, updateOtherUser } from '../../services/user'
import { clearNotifications } from '../../reducers/notificationReducer'
import { usePersistedState } from '../../hooks/persistedState'
import { AdminCourseList } from './AdminPage/AdminCourseList'
import { AdminUserList } from './AdminPage/AdminUserList'
import DocumentTitle from '../DocumentTitle'
import Error from '../Error'

export const AdminPage = props => {
  const state = usePersistedState('AdminPage', {
    coursesList: false,
    usersList: false
  })

  useEffect(() => {
    props.clearNotifications()
  }, [])

  useEffect(() => {
    if (state.coursesList) {
      props.resetLoading()
      props.getAllCI()
    }
  }, [state.coursesList])

  useEffect(() => {
    if (state.usersList) {
      props.resetLoading()
      props.getAllUsers()
    }
  }, [state.usersList])

  const documentTitle = <DocumentTitle title={'Admin'} />

  if (!props.user.user.sysop) {
    return <Redirect to={`/labtool`} />
  } else if (props.errors && props.errors.length > 0) {
    return <Error errors={props.errors.map(error => `${error.response.data} (${error.response.status} ${error.response.statusText})`)} />
  } else if (props.loading.loading) {
    return (
      <>
        <Loader active />
        {documentTitle}
      </>
    )
  }

  return (
    <>
      {documentTitle}
      <div className="AdminPage">
        <h2>Welcome to the admin interface.</h2>
        <Button compact className={`ui button`} onClick={() => (state.coursesList = !state.coursesList)}>
          {state.coursesList ? 'Hide courses' : 'Manage courses'}
        </Button>
        <br />
        <br />
        {state.coursesList && <AdminCourseList courses={props.courseInstance} />}
        <br />
        <hr />
        <br />
        <Button compact className={`ui button`} onClick={() => (state.usersList = !state.usersList)}>
          {state.usersList ? 'Hide users' : 'Manage users'}
        </Button>
        <br />
        {state.usersList && <AdminUserList me={props.user} users={props.users} updateOtherUser={props.updateOtherUser} />}
      </div>
    </>
  )
}

const mapStateToProps = state => {
  return {
    user: state.user,
    courseInstance: state.courseInstance,
    users: state.users,
    loading: state.loading,
    errors: Object.values(state.loading.errors)
  }
}

const mapDispatchToProps = {
  getAllCI,
  resetLoading,
  getAllUsers,
  updateOtherUser,
  clearNotifications
}

AdminPage.propTypes = {
  user: PropTypes.object.isRequired,
  courseInstance: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  users: PropTypes.array.isRequired,
  loading: PropTypes.object.isRequired,

  getAllCI: PropTypes.func.isRequired,
  resetLoading: PropTypes.func.isRequired,
  getAllUsers: PropTypes.func.isRequired,
  updateOtherUser: PropTypes.func.isRequired,
  clearNotifications: PropTypes.func.isRequired,

  errors: PropTypes.array
}

export default connect(mapStateToProps, mapDispatchToProps)(AdminPage)
