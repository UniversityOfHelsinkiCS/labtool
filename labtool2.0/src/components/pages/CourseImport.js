import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { Button, Header, Table, Loader, Form } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { Link, Redirect } from 'react-router-dom'
import { getIsAllowedToImport, getImportableCourses, importCourses } from '../../services/courseImport'
import { resetLoading, addRedirectHook } from '../../reducers/loadingReducer'
import { HorizontalScrollable } from '../HorizontalScrollable'
import { formatCourseName } from '../../util/format'

/**
 *  Show all the courses in a single list.
 */
export const CourseImport = props => {
  useEffect(() => {
    // run on component mount
    props.resetLoading()
    props.getIsAllowedToImport()
    props.getImportableCourses()
  }, [])

  const handleSubmit = async e => {
    e.preventDefault()
    const data = props.importable
    if (data) {
      const courses = data.filter(entry => e.target['import' + entry.hid] && e.target['import' + entry.hid].checked)
      props.addRedirectHook({
        hook: 'COURSE_IMPORT_DO_IMPORT_'
      })
      await props.importCourses({ courses })
    }
  }

  const renderCourse = instance => (
    <Table.Row key={instance.hid}>
      <Table.Cell>
        <Form.Checkbox name={'import' + instance.hid} />
      </Table.Cell>
      <Table.Cell>{instance.shorterId}</Table.Cell>
      <Table.Cell>{formatCourseName(instance.cname, instance.hid, instance.starts)}</Table.Cell>
      <Table.Cell>{instance.instructor}</Table.Cell>
      <Table.Cell>{instance.europeanStart}</Table.Cell>
      <Table.Cell>{instance.europeanEnd}</Table.Cell>
    </Table.Row>
  )

  return (
    <div className="CourseImport">
      <div className="mainContainer">
        <Header as="h2">Import courses</Header>
        {props.loading.loading || !Array.isArray(props.importable) || !props.canImport ? (
          <Loader active />
        ) : props.loading.redirect ? (
          <Redirect to="/labtool/courses" />
        ) : props.importable.length ? (
          <Form onSubmit={handleSubmit}>
            <HorizontalScrollable>
              <Table singleLine color="yellow" style={{ overflowX: 'visible' }}>
                <Table.Header>
                  <Table.Row>
                    <Table.HeaderCell colSpan="1" key={-1}>
                      Import?
                    </Table.HeaderCell>
                    <Table.HeaderCell colSpan="1">Course ID</Table.HeaderCell>
                    <Table.HeaderCell colSpan="1">Course name</Table.HeaderCell>
                    <Table.HeaderCell colSpan="1">Instructor (original)</Table.HeaderCell>
                    <Table.HeaderCell colSpan="1">Course start date</Table.HeaderCell>
                    <Table.HeaderCell colSpan="1">Course end date</Table.HeaderCell>
                  </Table.Row>
                </Table.Header>

                <Table.Body>{props.importable.map(renderCourse)}</Table.Body>
              </Table>
            </HorizontalScrollable>

            <br />
            <br />

            <Button type="submit" className="ui green button" content="Import" />

            <br />
            <br />

            <Link to="/labtool/courses">
              <Button className="ui button" type="cancel">
                Cancel
              </Button>
            </Link>
          </Form>
        ) : (
          <div>
            <h4>There are no courses in Kurki to import</h4>

            <Link to="/labtool/courses">
              <Button className="ui button" type="cancel">
                Back
              </Button>
            </Link>
          </div>
        )}
      </div>
    </div>
  )
}

const mapStateToProps = (state, ownProps) => {
  return {
    ...ownProps,
    loading: state.loading,
    canImport: state.courseImport.canImport,
    importable: state.courseImport.importable
  }
}

const mapDispatchToProps = {
  resetLoading,
  getIsAllowedToImport,
  getImportableCourses,
  importCourses,
  addRedirectHook
}

CourseImport.propTypes = {
  loading: PropTypes.object.isRequired,
  canImport: PropTypes.bool,
  importable: PropTypes.array,

  resetLoading: PropTypes.func.isRequired,
  getIsAllowedToImport: PropTypes.func.isRequired,
  getImportableCourses: PropTypes.func.isRequired,
  importCourses: PropTypes.func.isRequired,
  addRedirectHook: PropTypes.func.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CourseImport)
