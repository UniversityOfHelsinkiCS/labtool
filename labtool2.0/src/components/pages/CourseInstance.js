import React from 'react'

const CourseInstance = ({ instance, handleFieldChange }) => {
  return(
    <li> {instance.name} <button onClick={handleFieldChange} name="courseInstanceId" value={instance.id}>Register</button>
    </li>
  )
}

export default CourseInstance