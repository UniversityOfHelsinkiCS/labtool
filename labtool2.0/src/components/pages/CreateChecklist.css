.deleteButton {
  float: right;
}

.addForm {
  margin: 0px 10px 10px 10px;
}

.numberField {
  width: 100px;
}

.textField {
  width: 100%;
}

.formField {
  margin-top: 10px;
}

.topOptions {
  margin-bottom: 15px;
}

.saveButtonText {
  font-size: 24px;
}

.deleteButtonText {
  font-size: 14px;
}