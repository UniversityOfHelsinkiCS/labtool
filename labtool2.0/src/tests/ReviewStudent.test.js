import React from 'react'
import { ReviewStudent } from '../components/pages/ReviewStudent'
import { shallow } from 'enzyme'

describe('<ReviewStudent />', () => {
  let wrapper

  const ownProps = {
    courseId: 'TKT20010.2018.K.A.1',
    studentInstance: '10011',
    weekNumber: '1'
  }

  const props = {
    selectedInstance: {
      id: 10011,
      name: 'Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit',
      start: '2018-03-11T21:00:00.000Z',
      end: '2018-04-29T21:00:00.000Z',
      active: true,
      weekAmount: 7,
      weekMaxPoints: 3,
      currentWeek: 1,
      ohid: 'TKT20010.2018.K.A.1',
      checklists: [
        {
          id: 10002,
          week: 2,
          list: {},
          courseName: 'Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit',
          master: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z',
          courseInstanceId: 10011
        },
        {
          id: 10003,
          week: 1,
          list: {
            Koodi: [
              {
                name: 'Koodin laatu',
                points: 2,
                textWhenOn: 'Koodi tehty laadukkaasti ja sisältää kommentteja',
                textWhenOff: 'Koodin laadussa parantamisen varaa'
              }
            ],
            Algoritmit: [
              {
                name: 'Algoritmin runko',
                points: 2,
                textWhenOn: 'Algoritmin runko luotu',
                textWhenOff: 'Algoritmin runko puuttuu'
              },
              {
                name: 'Tietorakenteita luotu',
                points: 2,
                textWhenOn: 'Tietorakenteita luotu',
                textWhenOff: 'Tietorakenteita ei ole luotu'
              }
            ],
            Dokumentaatio: [
              {
                name: 'Readme',
                points: 1,
                textWhenOn: 'README kunnossa',
                textWhenOff: 'README puuttuu'
              },
              {
                name: 'Tuntikirjanpito',
                points: 1,
                textWhenOn: 'Tuntikirjanpito täytetty oikein',
                textWhenOff: 'Tuntikirjanpito puuttuu'
              }
            ]
          },
          courseName: 'Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit',
          master: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z',
          courseInstanceId: 10011
        }
      ],
      createdAt: '2018-03-26T00:00:00.000Z',
      updatedAt: '2018-03-26T00:00:00.000Z',
      teacherInstances: [
        {
          id: 10001,
          admin: true,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z',
          userId: 10010,
          courseInstanceId: 10011,
          firsts: 'Pää',
          lastname: 'Opettaja'
        },
        {
          id: 10011,
          admin: true,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z',
          userId: 10015,
          courseInstanceId: 10011,
          firsts: 'Ossi Ohjaaja',
          lastname: 'Mutikainen'
        }
      ]
    },
    studentInstance: 10011,
    coursePage: {
      role: 'teacher',
      data: [
        {
          id: 10012,
          github: 'http://github.com/tiralabra2',
          projectName: 'Tiran toinen labraprojekti',
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z',
          courseInstanceId: 10011,
          userId: 10012,
          teacherInstanceId: 10011,
          weeks: [],
          codeReviews: [],
          User: {
            id: 10012,
            username: 'tiraopiskelija2',
            email: 'johan.studerande@helsinki.fi',
            firsts: 'Johan Wilhelm',
            lastname: 'Studerande',
            studentNumber: '014553242',
            admin: false,
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z'
          }
        },
        {
          id: 10031,
          github: 'http://github.com/superprojekti',
          projectName: 'Tira super projekti',
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-06-05T07:12:28.603Z',
          courseInstanceId: 10011,
          userId: 10031,
          teacherInstanceId: 10011,
          weeks: [],
          codeReviews: [],
          User: {
            id: 10031,
            username: 'superopiskelija',
            email: 'teras.henkilo@helsinki.fi',
            firsts: 'Teräs',
            lastname: 'Henkilö',
            studentNumber: '014666666',
            admin: false,
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z'
          }
        },
        {
          id: 10011,
          github: 'http://github.com/tiralabra1',
          projectName: 'Tiran labraprojekti',
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z',
          courseInstanceId: 10011,
          userId: 10011,
          teacherInstanceId: 10011,
          weeks: [
            {
              id: 10002,
              points: 2,
              weekNumber: 2,
              feedback: 'Melko hienoa työtä!',
              createdAt: '2018-03-26T00:00:00.000Z',
              updatedAt: '2018-03-26T00:00:00.000Z',
              studentInstanceId: 10011,
              comments: []
            },
            {
              id: 10001,
              points: 3,
              weekNumber: 1,
              feedback: 'Hienoa työtä!',
              createdAt: '2018-03-26T00:00:00.000Z',
              updatedAt: '2018-03-26T00:00:00.000Z',
              studentInstanceId: 10011,
              comments: []
            },
            {
              id: 10003,
              points: 3,
              weekNumber: 3,
              feedback: 'Erittäin hienoa työtä!',
              createdAt: '2018-03-26T00:00:00.000Z',
              updatedAt: '2018-03-26T00:00:00.000Z',
              studentInstanceId: 10011,
              comments: []
            },
            {
              id: 10004,
              points: 3,
              weekNumber: 4,
              feedback: 'Hyvin menee!',
              createdAt: '2018-03-26T00:00:00.000Z',
              updatedAt: '2018-03-26T00:00:00.000Z',
              studentInstanceId: 10011,
              comments: []
            }
          ],
          codeReviews: [],
          User: {
            id: 10011,
            username: 'tiraopiskelija1',
            email: 'maarit.opiskelija@helsinki.fi',
            firsts: 'Maarit Mirja',
            lastname: 'Opiskelija',
            studentNumber: '014578343',
            admin: false,
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z'
          }
        }
      ]
    },
    weekReview: {
      data: [
        {
          id: 10031,
          github: 'http://github.com/superprojekti',
          projectName: 'Tira super projekti',
          courseInstanceId: 10011,
          userId: 10031,
          teacherInstanceId: 10001,
          weeks: [
            {
              id: 1,
              points: 0,
              weekNumber: 1,
              feedback: 'Koodin laadussa parantamisen varaa\n\nAlgoritmin runko puuttuu\n\nTietorakenteita ei ole luotu\n\nREADME puuttuu\n\nTuntikirjanpito puuttuu\n\n',
              notified: false,
              checks: {
                'Algoritmin runko': true,
                'Tietorakenteita luotu': true,
                'Koodin laatu': true
              },
              studentInstanceId: 10031,
              comments: []
            }
          ],
          codeReviews: [
            {
              id: 10003,
              points: null,
              reviewNumber: 1,
              linkToReview: null,
              studentInstanceId: 10031,
              toReview: 10011
            },
            {
              id: 10008,
              points: null,
              reviewNumber: 2,
              linkToReview: null,
              studentInstanceId: 10031,
              toReview: 10012
            }
          ],
          User: {
            id: 10031,
            username: 'superopiskelija',
            email: 'teras.henkilo@helsinki.fi',
            firsts: 'Teräs',
            lastname: 'Henkilö',
            studentNumber: '014666666',
            admin: false
          },
          Tags: [
            {
              id: 20007,
              name: 'FORTRAN',
              color: 'pink',
              StudentTag: {
                id: 30007,
                studentInstanceId: 10031,
                tagId: 20007,
                createdAt: '2018-06-13T00:00:00.000Z',
                updatedAt: '2018-06-13T00:00:00.000Z'
              }
            }
          ]
        },
        {
          id: 10011,
          github: 'http://github.com/tiraopi1/tiralabra1',
          projectName: 'Tiran labraprojekti',
          courseInstanceId: 10011,
          userId: 10011,
          teacherInstanceId: 10011,
          weeks: [
            {
              id: 10002,
              points: 2,
              weekNumber: 2,
              feedback: 'Melko hienoa työtä!',
              notified: false,
              checks: {},
              studentInstanceId: 10011,
              comments: []
            },
            {
              id: 10001,
              points: 3,
              weekNumber: 1,
              feedback: 'Hienoa työtä!',
              notified: false,
              checks: {},
              studentInstanceId: 10011,
              comments: []
            },
            {
              id: 10003,
              points: 3,
              weekNumber: 3,
              feedback: 'Erittäin hienoa työtä!',
              notified: false,
              checks: {},
              studentInstanceId: 10011,
              comments: []
            },
            {
              id: 10004,
              points: 3,
              weekNumber: 4,
              feedback: 'Hyvin menee!',
              notified: false,
              checks: {},
              studentInstanceId: 10011,
              comments: []
            }
          ],
          codeReviews: [
            {
              id: 10001,
              points: null,
              reviewNumber: 1,
              linkToReview: null,
              studentInstanceId: 10011,
              toReview: 10012
            },
            {
              id: 10007,
              points: null,
              reviewNumber: 2,
              linkToReview: null,
              studentInstanceId: 10011,
              toReview: 10031
            }
          ],
          User: {
            id: 10011,
            username: 'tiraopiskelija1',
            email: 'maarit.opiskelija@helsinki.fi',
            firsts: 'Maarit Mirja',
            lastname: 'Opiskelija',
            studentNumber: '014578343',
            admin: false
          },
          Tags: [
            {
              id: 20003,
              name: 'game',
              color: 'black',
              StudentTag: {
                id: 30003,
                studentInstanceId: 10011,
                tagId: 20003,
                createdAt: '2018-06-13T00:00:00.000Z',
                updatedAt: '2018-06-13T00:00:00.000Z'
              }
            },
            {
              id: 20002,
              name: 'HTML',
              color: 'yellow',
              StudentTag: {
                id: 30002,
                studentInstanceId: 10011,
                tagId: 20002,
                createdAt: '2018-06-13T00:00:00.000Z',
                updatedAt: '2018-06-13T00:00:00.000Z'
              }
            },
            {
              id: 20001,
              name: 'Javascript',
              color: 'red',
              StudentTag: {
                id: 30001,
                studentInstanceId: 10011,
                tagId: 20001,
                createdAt: '2018-06-13T00:00:00.000Z',
                updatedAt: '2018-06-13T00:00:00.000Z'
              }
            }
          ]
        },
        {
          id: 10012,
          github: 'http://github.com/tiraopi2/tiralabra2',
          projectName: 'Tiran toinen labraprojekti',
          courseInstanceId: 10011,
          userId: 10012,
          teacherInstanceId: 10011,
          weeks: [],
          codeReviews: [
            {
              id: 10002,
              points: null,
              reviewNumber: 1,
              linkToReview: null,
              studentInstanceId: 10012,
              toReview: 10031
            },
            {
              id: 10009,
              points: null,
              reviewNumber: 2,
              linkToReview: null,
              studentInstanceId: 10012,
              toReview: 10011
            }
          ],
          User: {
            id: 10012,
            username: 'tiraopiskelija2',
            email: 'johan.studerande@helsinki.fi',
            firsts: 'Johan Wilhelm',
            lastname: 'Studerande',
            studentNumber: '014553242',
            admin: false
          },
          Tags: [
            {
              id: 20005,
              name: 'Node.js',
              color: 'blue',
              StudentTag: {
                id: 30005,
                studentInstanceId: 10012,
                tagId: 20005,
                createdAt: '2018-06-13T00:00:00.000Z',
                updatedAt: '2018-06-13T00:00:00.000Z'
              }
            },
            {
              id: 20001,
              name: 'Javascript',
              color: 'red',
              StudentTag: {
                id: 30004,
                studentInstanceId: 10012,
                tagId: 20001,
                createdAt: '2018-06-13T00:00:00.000Z',
                updatedAt: '2018-06-13T00:00:00.000Z'
              }
            }
          ]
        }
      ],
      checks: {}
    },
    loading: {
      loading: false,
      loadingHooks: [],
      redirect: false,
      redirectHooks: [],
      redirectFailure: false
    }
  }

  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <ReviewStudent
        getOneCI={mockFn}
        clearNotifications={mockFn}
        courseData={props.coursePage}
        ownProps={ownProps}
        selectedInstance={props.selectedInstance}
        weekReview={props.weekReview}
        loading={props.loading}
        resetLoading={mockFn}
        coursePageInformation={mockFn}
      />
    )
  })

  describe('ReviewStudent Component', () => {
    it('is ok', () => {
      true
    })

    it('should render without throwing an error', () => {
      expect(wrapper.find('.ReviewStudent').exists()).toEqual(true)
    })

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    describe('Checklist', () => {
      it('renders a checklist', () => {
        expect(wrapper.find('.checklist').exists()).toEqual(true)
      })

      it('renders a card for each checklist topic', () => {
        const expected = Object.keys(props.selectedInstance.checklists.find(cl => cl.week === props.selectedInstance.currentWeek).list).length
        expect(wrapper.find('.checklistCard').length).toEqual(expected)
      })

      it('renders a row for each checklist item', () => {
        let expected = 0
        const checklist = props.selectedInstance.checklists.find(cl => cl.week === props.selectedInstance.currentWeek).list
        Object.keys(checklist).forEach(key => {
          expected += checklist[key].length
        })
        expect(wrapper.find('.checklistCardRow').length).toEqual(expected)
      })
    })
  })
})
