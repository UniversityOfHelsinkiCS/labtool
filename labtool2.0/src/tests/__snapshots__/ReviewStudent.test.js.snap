// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewStudent /> ReviewStudent Component should render correctly 1`] = `
<div
  className="ReviewStudent"
  style={
    Object {
      "textAlign": "center",
      "textAlignVertical": "center",
    }
  }
>
  <h2>
     
    Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit
  </h2>
  <h3>
     
    Maarit Mirja
     
    Opiskelija
     
    <div
      style={
        Object {
          "display": "inline-block",
          "padding": "0px 0px 0px 25px",
        }
      }
    >
      Tiran labraprojekti
      : 
      <RepoLink
        url="http://github.com/tiraopi1/tiralabra1"
      />
    </div>
    <div
      key="20003"
    >
      <Button
        as="button"
        className="mini ui black button"
        compact={true}
        floated="right"
      >
        game
      </Button>
    </div>
    <div
      key="20002"
    >
      <Button
        as="button"
        className="mini ui yellow button"
        compact={true}
        floated="right"
      >
        HTML
      </Button>
    </div>
    <div
      key="20001"
    >
      <Button
        as="button"
        className="mini ui red button"
        compact={true}
        floated="right"
      >
        Javascript
      </Button>
    </div>
  </h3>
  <h3>
    Week 
  </h3>
  <Grid>
    <GridRow
      columns={2}
    >
      <GridColumn>
        <div
          align="left"
        >
          <h3>
            Points from previous weeks: 
            0
             
          </h3>
          Week points: 
          0
           
          <br />
          Code review points: 
          0
        </div>
        <PreviousWeekDetails />
        <h2>
          Review
        </h2>
        <Form
          as="form"
          onSubmit={[Function]}
        >
          <FormGroup
            inline={true}
            unstackable={true}
          >
            <FormField>
              <label
                className="showMaxPoints"
              >
                Points 0-
                4
              </label>
              <Input
                name="points"
                onChange={[Function]}
                step="0.01"
                style={
                  Object {
                    "align": "center",
                    "width": "150px",
                  }
                }
                type="number"
                value={3}
              />
            </FormField>
          </FormGroup>
          <h4>
            Feedback
          </h4>
          <FormGroup
            inline={true}
            style={
              Object {
                "textAlignVertical": "top",
              }
            }
            unstackable={true}
          >
            <div
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <FormMarkdownTextArea
                name="comment"
                onChange={[Function]}
                style={
                  Object {
                    "height": "250px",
                    "width": "500px",
                  }
                }
                value="Hienoa työtä!"
              />
            </div>
          </FormGroup>
          <h4>
            Review notes
          </h4>
          <p>
            <em>
              Only shown to instructors on this course
            </em>
          </p>
          <FormGroup
            inline={true}
            style={
              Object {
                "textAlignVertical": "top",
              }
            }
            unstackable={true}
          >
            <div
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <FormMarkdownTextArea
                name="instructorNotes"
                onChange={[Function]}
                style={
                  Object {
                    "height": "150px",
                    "width": "500px",
                  }
                }
              />
            </div>
          </FormGroup>
          <FormField>
            <Button
              as="button"
              className="ui center floated green button"
              type="submit"
            >
              Save
            </Button>
            <Button
              as="button"
              className="ui center floated button"
              onClick={[Function]}
              type="button"
            >
              Save as draft
            </Button>
            <Link
              to="/labtool/browsereviews/TKT20010.2018.K.A.1/10011"
              type="Cancel"
            >
              <Button
                as="button"
                className="ui center floated button"
                onClick={[Function]}
                type="cancel"
              >
                Cancel
              </Button>
            </Link>
          </FormField>
        </Form>
      </GridColumn>
      <GridColumn>
        <h2>
          Checklist
        </h2>
        <div
          className="checklist"
        >
          <Card
            className="checklistCard"
            color="red"
            fluid={true}
            key="Koodi"
          >
            <CardContent
              header="Koodi"
            />
            <CardContent
              className="checklistCardRow"
              key="Koodin laatu"
              onClick={[Function]}
            >
              <FormField>
                <Grid>
                  <GridRow
                    style={
                      Object {
                        "cursor": "pointer",
                        "userSelect": "none",
                      }
                    }
                  >
                    <GridColumn
                      width={3}
                    >
                      <Icon
                        as="i"
                        name="circle outline"
                        size="large"
                        style={
                          Object {
                            "color": "black",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      width={10}
                    >
                      <span
                        style={
                          Object {
                            "flexGrow": 1,
                            "textAlign": "center",
                          }
                        }
                      >
                        Koodin laatu
                      </span>
                    </GridColumn>
                    <GridColumn
                      width={3}
                    >
                      <span>
                        undefined p / undefined p
                      </span>
                    </GridColumn>
                  </GridRow>
                </Grid>
              </FormField>
            </CardContent>
          </Card>
          <Card
            className="checklistCard"
            color="red"
            fluid={true}
            key="Algoritmit"
          >
            <CardContent
              header="Algoritmit"
            />
            <CardContent
              className="checklistCardRow"
              key="Algoritmin runko"
              onClick={[Function]}
            >
              <FormField>
                <Grid>
                  <GridRow
                    style={
                      Object {
                        "cursor": "pointer",
                        "userSelect": "none",
                      }
                    }
                  >
                    <GridColumn
                      width={3}
                    >
                      <Icon
                        as="i"
                        name="circle outline"
                        size="large"
                        style={
                          Object {
                            "color": "black",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      width={10}
                    >
                      <span
                        style={
                          Object {
                            "flexGrow": 1,
                            "textAlign": "center",
                          }
                        }
                      >
                        Algoritmin runko
                      </span>
                    </GridColumn>
                    <GridColumn
                      width={3}
                    >
                      <span>
                        undefined p / undefined p
                      </span>
                    </GridColumn>
                  </GridRow>
                </Grid>
              </FormField>
            </CardContent>
            <CardContent
              className="checklistCardRow"
              key="Tietorakenteita luotu"
              onClick={[Function]}
            >
              <FormField>
                <Grid>
                  <GridRow
                    style={
                      Object {
                        "cursor": "pointer",
                        "userSelect": "none",
                      }
                    }
                  >
                    <GridColumn
                      width={3}
                    >
                      <Icon
                        as="i"
                        name="circle outline"
                        size="large"
                        style={
                          Object {
                            "color": "black",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      width={10}
                    >
                      <span
                        style={
                          Object {
                            "flexGrow": 1,
                            "textAlign": "center",
                          }
                        }
                      >
                        Tietorakenteita luotu
                      </span>
                    </GridColumn>
                    <GridColumn
                      width={3}
                    >
                      <span>
                        undefined p / undefined p
                      </span>
                    </GridColumn>
                  </GridRow>
                </Grid>
              </FormField>
            </CardContent>
          </Card>
          <Card
            className="checklistCard"
            color="red"
            fluid={true}
            key="Dokumentaatio"
          >
            <CardContent
              header="Dokumentaatio"
            />
            <CardContent
              className="checklistCardRow"
              key="Readme"
              onClick={[Function]}
            >
              <FormField>
                <Grid>
                  <GridRow
                    style={
                      Object {
                        "cursor": "pointer",
                        "userSelect": "none",
                      }
                    }
                  >
                    <GridColumn
                      width={3}
                    >
                      <Icon
                        as="i"
                        name="circle outline"
                        size="large"
                        style={
                          Object {
                            "color": "black",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      width={10}
                    >
                      <span
                        style={
                          Object {
                            "flexGrow": 1,
                            "textAlign": "center",
                          }
                        }
                      >
                        Readme
                      </span>
                    </GridColumn>
                    <GridColumn
                      width={3}
                    >
                      <span>
                        undefined p / undefined p
                      </span>
                    </GridColumn>
                  </GridRow>
                </Grid>
              </FormField>
            </CardContent>
            <CardContent
              className="checklistCardRow"
              key="Tuntikirjanpito"
              onClick={[Function]}
            >
              <FormField>
                <Grid>
                  <GridRow
                    style={
                      Object {
                        "cursor": "pointer",
                        "userSelect": "none",
                      }
                    }
                  >
                    <GridColumn
                      width={3}
                    >
                      <Icon
                        as="i"
                        name="circle outline"
                        size="large"
                        style={
                          Object {
                            "color": "black",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      width={10}
                    >
                      <span
                        style={
                          Object {
                            "flexGrow": 1,
                            "textAlign": "center",
                          }
                        }
                      >
                        Tuntikirjanpito
                      </span>
                    </GridColumn>
                    <GridColumn
                      width={3}
                    >
                      <span>
                        undefined p / undefined p
                      </span>
                    </GridColumn>
                  </GridRow>
                </Grid>
              </FormField>
            </CardContent>
          </Card>
          <div>
            <Form
              as="form"
              className="checklistOutput"
              onSubmit={[Function]}
            >
              <FormTextArea
                as={[Function]}
                className="checklistOutputText"
                control={[Function]}
                name="text"
                style={
                  Object {
                    "height": "250px",
                    "width": "100%",
                  }
                }
                value="Koodin laadussa parantamisen varaa

Algoritmin runko puuttuu

Tietorakenteita ei ole luotu

README puuttuu

Tuntikirjanpito puuttuu

"
              />
              <p
                className="checklistOutputPoints"
              >
                points: 
                NaN
              </p>
              <input
                name="points"
                type="hidden"
                value={NaN}
              />
              <Button
                as="button"
                type="submit"
              >
                Copy to review fields
              </Button>
            </Form>
          </div>
        </div>
      </GridColumn>
    </GridRow>
  </Grid>
</div>
`;
